apps/src/applab/dropletConfig.js

```diff
@@ -149,14 +149,14 @@ module.exports.blocks = [
 
   {func: 'console.log', parent: consoleApi, category: 'Variables', paletteParams: ['message'], params: ['"message"'] },
   {func: 'declareAssign_str_hello_world', block: 'var str = "Hello World";', category: 'Variables', noAutocomplete: true },
-  {func: 'substring', blockPrefix: 'str.substring', category: 'Variables', paletteParams: ['start','end'], params: ["6", "11"], modeOptionName: '*.substring', type: 'value' },
-  {func: 'indexOf', blockPrefix: 'str.indexOf', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.indexOf', type: 'value' },
-  {func: 'includes', blockPrefix: 'str.includes', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.includes', type: 'value' },
-  {func: 'length', block: 'str.length', category: 'Variables', modeOptionName: '*.length', type: 'property' },
-  {func: 'toUpperCase', blockPrefix: 'str.toUpperCase', category: 'Variables', modeOptionName: '*.toUpperCase', type: 'value' },
-  {func: 'toLowerCase', blockPrefix: 'str.toLowerCase', category: 'Variables', modeOptionName: '*.toLowerCase', type: 'value' },
+  {func: 'substring', blockPrefix: 'str.', category: 'Variables', paletteParams: ['start','end'], params: ["6", "11"], modeOptionName: '*.substring', tipPrefix: '[string].', type: 'value' },
+  {func: 'indexOf', blockPrefix: 'str.', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.indexOf', tipPrefix: '[string].', type: 'value' },
+  {func: 'includes', blockPrefix: 'str.', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.includes', tipPrefix: '[string].', type: 'value' },
+  {func: 'length', block: 'str.length', category: 'Variables', modeOptionName: '*.length', tipPrefix: '[string].', type: 'property' },
+  {func: 'toUpperCase', blockPrefix: 'str.', category: 'Variables', modeOptionName: '*.toUpperCase', tipPrefix: '[string].', type: 'value' },
+  {func: 'toLowerCase', blockPrefix: 'str.', category: 'Variables', modeOptionName: '*.toLowerCase', tipPrefix: '[string].', type: 'value' },
   {func: 'declareAssign_list_abd', block: 'var list = ["a", "b", "d"];', category: 'Variables', noAutocomplete: true },
-  {func: 'listLength', block: 'list.length', category: 'Variables', noAutocomplete: true, type: 'property' },
+  {func: 'listLength', block: 'list.length', category: 'Variables', noAutocomplete: true, tipPrefix: '[list].', type: 'property' },
```

Comment:
Local variables is a good idea to avoid repeating. Although there may be situations where it is used for some localizable scenario, for now, it is only being used to put a type in brackets in the tooltip, which should not be localized. Previously, this was solved by creating a localized string with the suffix `_signatureOverride`, which was a one-size-fits-all approach for solving one-offs. But in an object oriented API, this isn't really a one-off, so we need a better, data-driven solution.